### PR TITLE
[ENH] Add support for sklearn HistGradientBoostingEstimator

### LINKED ISSUE #868: HistGradientBoosting support
Right now it could be only added by writing custom class.

Question is: will there be added support to `flamlize_estimator` for `HistGradientBoostingClassifier` and `HistGradientBoostingRegressor` ?
```[tasklist]
### Tasks
- [x] Implement a custom estimator following
- [x] Add test to make sure it works as intended
- [x] Make a subpackage automl.contrib and include the newly created estimators in it
- [x] Add a section "contributed estimators" in the documentation website: https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML#estimator
- [x] Change the test to use the newly added estimators
```

### PR DESCRIPTION
**Pull request by glevv**
## Why are these changes needed?

This PR Adds support for HistGradientBoosting estimator

## Related issue number

Closes #868

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by glevv** [comment]
@microsoft-github-policy-service agree

**Comment by glevv** [comment]
pre-commit hook made a lot of unnecessary changes

**Comment by glevv** [comment]
Not sure about how should I fix this empty dataframe error

**Comment by sonichi** [comment]
The test passed. @qingyun-wu could you review or find other reviewers?

File: flaml/automl/model.py [hunk 1]
```diff
@@ -2,36 +2,42 @@
...
-import os
 from typing import Callable, List, Union
+
 import numpy as np
-import time
-import logging
-import shutil
-import sys
-import math
+
```
Comment (thinkall) [inline flaml/automl/model.py:17]: Could we avoid these kind of changes? In fact, I like the new format, but would prefer use a new PR just for formatting.

**Review by thinkall — COMMENTED** [review]
Thank you @glevv for the contribution, nice PR!

File: flaml/automl/model.py (hunk 1 above)
Comment (thinkall) [reply flaml/automl/model.py:17]: > Yes, I did not want to do it but pre-commit made all these changes. Maybe I configured it wrong. Should I try to revert it?

I hope so, if you can do that without too much trouble. I tried to clone the repo into a new folder and run precommit to all files, nothing changed. So I guess maybe you've different configs. Given this, next time a new PR is coming, I don't know if any changes in this PR will be changed back again (high chance not, but I'm not sure).

Comment (thinkall) [reply flaml/automl/model.py:17]: what do you think on this format thing? @sonichi , @qingyun-wu

Comment (sonichi) [reply flaml/automl/model.py:17]: @Borda  do you have any suggestion?

Comment (glevv) [reply flaml/automl/model.py:17]: Yep, I think changes were made by vscode-isort / isort library, I think it should be possible to revert it without too much of a hassle

Comment (sonichi) [reply flaml/automl/model.py:17]: I think the format change is OK. We can keep them.

Comment (glevv) [reply flaml/automl/model.py:17]: I think it's doable to `isort` the whole library, add it to pre-commit hook with args `--profile black` and add it to dev dependencies.

**Review by thinkall — APPROVED** [review]

**Review by sonichi — APPROVED** [review]

**Pull request merged**